### DISCUSSION TITLE
Innlux und Grafana

### DISCUSSION THREAD

**Discussion by Deanw1975** [solectrus/solectrus#2924] [Q&A]
Hallo zusammen,

ich habe die Solectrus INFLUX für den Webzugang geöffnet.
Der Zugriff und Abfrage über Webzugang funktioniert:

![c93d9cf1fb304f4287d543e74064ae8eaa0b3c37](https://github.com/solectrus/solectrus/assets/108386811/4427b867-6157-4c95-8e5c-4725da8112be)

In Grafana habe ich die DB auch eingerichtet:
![grafana](https://github.com/solectrus/solectrus/assets/108386811/e3536653-7682-417a-8920-51f227f7a51f)

![grafana2](https://github.com/solectrus/solectrus/assets/108386811/798b5692-b8f0-46e5-af94-35124c5d143f)

Wenn ich in Grafen auf "Save & Test" klicke bekomme ich einen positive Rückmeldung.

In Grafana versuche ich dann auf die Solectrus DB eine Abfrage durchzuführen  und sehe unter "where" in den Feldern 2 mal Doppelpunkt:

![doppelpunkt](https://github.com/solectrus/solectrus/assets/108386811/c1ac42c7-99c1-451c-8658-97685f1c49b7)

Kann also die Abfrage auf die Hauseigene Solectrus DB nicht in Grafana machen sondern nur auf der Weboberfläche der Influx-DB direkt.

Wo liegt mein Fehler?

LG
Dean

**Comment by ledermann** [comment]
Mit Grafana habe ich bislang nichts gemacht, da kenne ich mich leider gar nicht mit aus.

Soweit ich gerade recherchiert habe, zeigt Grafana an dieser Stelle Fields oder Tags an. In der InfluxDB von SOLECTRUS gibt es keine Tags, es gibt nur Measurement und Fields. Insofern sieht das für mich nicht so falsch aus. Funktioniert es denn nicht, wenn man ein solch dekoriertes Field auswählt?

**Comment by Deanw1975** [comment]
Danke für die Antwort.
Wie bekomme ich die Tags in die Solectrus Influx?
Hat da jemand eine Idee?

VG
Dean

**Reply by ledermann** [reply]
Die Struktur der Messwerte von SOLECTRUS ist so simpel, dass bislang keine Tags verwendet werden.

Tags sind Metadaten zu Messwerten. Wenn ich zum Beispiel Messwerte zur Temperatur sammele, dann könnte mittels Tags der Messpunkt angegeben werden, also z.B. "Küche" oder "Wohnzimmer". Diese Art von Metadaten macht vor allem Sinn, wenn man viele Messwerte sammelt. Das macht SOLECTRUS aber gar nicht.

Ein nachträgliches Ergänzen von Tags ist auch nicht vorgesehen. Tags müssten direkt beim Schreiben des Messwertes vergeben werden, die fügt man nicht nachträglich hinzu.

Vielleicht prüfst du mal, ob du überhaupt Tags benötigst. Eine InfluxDB-Datenbank, die ohne Tags auskommt, ist nichts Ungewöhnliches.

**Reply by Deanw1975** [reply]
Guten Morgen Georg,

Danke für die Mühe. Ich werde mal anderweitig Hilfe suchen.

Lasse bitte die Anfrage offen, evtl. gibt es noch jemanden der eine Idee hat.

VG
Dean
